src/pip/_internal/req/req_uninstall.py

```diff
@@ -544,8 +544,8 @@ def from_dist(cls, dist):
         elif develop_egg_link:
             # develop egg
             with open(develop_egg_link, 'r') as fh:
-                link_pointer = os.path.normcase(fh.readline().strip())
-            assert (link_pointer == dist.location), (
+                link_pointer = normalize_path(os.path.normcase(fh.readline().strip()))
+            assert (link_pointer == dist_path), (
```

Comment:
At this point, it’s probably better to use `Path.resolve(strict=False)` instead. These various path string functions should be removed eventually.

Reply:
I don't have so much into the internals of `pip` so I really am not the best person to comment. Anyway, I did some testing on the [`Path.resolve`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve) with a folder that is also symlinked on Windows. `C:\cases\myproj` points to `C:\Users\niko\Dropbox\myproj`.

```python
In [1]: from pathlib import Path

In [2]: mypath = Path(r'C:\cases\myproj')

In [3]: mypath.resolve()
Out[3]: WindowsPath('C:/Users/niko/Dropbox/myproj')

In [4]: mypath.resolve(strict=True)
Out[4]: WindowsPath('C:/Users/niko/Dropbox/myproj')

In [5]: mypath.resolve(strict=False)
Out[5]: WindowsPath('C:/Users/niko/Dropbox/myproj')
```

Did not see any difference with `strict=True` or `strict=False`. Which lines / objects you would change to use `Path.resolve`?